packages/clerk-js/src/ui/hooks/useDirection.ts

```diff
@@ -0,0 +1,32 @@
+function getDirectionFromElement(element: HTMLElement): 'ltr' | 'rtl' {
+  const dir = element.dir;
+
+  if (dir === 'rtl') {
+    return 'rtl';
+  }
+
+  if (dir === 'ltr') {
+    return 'ltr';
+  }
+
+  if (dir === 'auto' || !dir) {
+    const computedDirection = window.getComputedStyle(element).direction;
+    if (computedDirection === 'rtl') {
+      return 'rtl';
+    }
+  }
+
+  return 'ltr';
```

Comment:
can it be simplified to this ?
```suggestion
  const default = 'ltr'
  if (dir === 'auto' || !dir) {
    const computedDirection = window.getComputedStyle(element).direction;
    return computedDirection || default;
  }

  return dir || default;
```